File: src/main/java/com/sliusar/todolist/services/TodoService.java
package com.sliusar.todolist.services;

import com.sliusar.todolist.models.Todo;

import java.util.List;
import java.util.Optional;

public interface TodoService {
    List<Todo> getAll();
    Optional<Todo> getById(int id);
    void save(Todo todo);
    void update(Todo todo, int todoId);
    void deleteById(int id);
    void updateTodo(Todo todo, int scheduleId);

    void addTodo(Todo todo, int scheduleId);
    void removeTodo(int todoId, int scheduleId);

}

File: src/main/frontend/src/app/services/todo.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Todo} from "../models/todo";
import {Observable} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class TodoService {

  URL = "http://localhost:8080/todoItems"
  constructor(private httpClient: HttpClient) { }

  update(todo: Todo, scheduleId: number): Observable<any>{
    return this.httpClient.put<Todo>(`${this.URL}/${scheduleId}/update/${todo.id}`, todo)
  }

  getAll(): Observable<any>{
    return this.httpClient.get<Todo>(this.URL)
  }

  getById(id: number): Observable<any>{
    return this.httpClient.get<any>(`${this.URL}/${id}`)
  }

  addTodo(todo: Todo, scheduleId: any): Observable<any>{
    return this.httpClient.post<Todo>(`${this.URL}/${scheduleId}/addTodo`, todo)
  }

  deleteTodo(scheduleId: any, todoId: number): Observable<any>{
    return this.httpClient.delete<any>(`${this.URL}/${scheduleId}/delete/${todoId}`)
  }
}
